### PR TITLE
Add client for interacting with resources API

### PR DESCRIPTION
**Pull request by newswangerd**

### CODE REVIEW

File: ansible_base/resource_registry/rest_client.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        """
+
+        self.base_url = f"{service_url}/{service_path.strip('/')}/"
+        self.requests_auth_kwargs = requests_auth_kwargs
+
+    def _make_request(self, method: str, path: str, data: dict = None) -> requests.Response:
+        url = self.base_url + path.lstrip("/")
+        logger.info(f"Making {method} request to {url}.")
+
+        kwargs = {**self.requests_auth_kwargs, "method": method, "url": url, "verify": False}
```
[the remark this answers is not in the document]
Comment (newswangerd) [reply ansible_base/resource_registry/rest_client.py]: I fixed this, go away

File: ansible_base/resource_registry/rest_client.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+logger = logging.getLogger('ansible_base.resources_api.rest_client')
+
+
+class ResourceAPIClient:
+    """
+    Client for Ansible services to interact with the service-index/ api
+    """
+
+    ResourceRequestBody = ResourceRequestBody
```
Comment (relrod) [inline ansible_base/resource_registry/rest_client.py]: :face_with_spiral_eyes:

File: ansible_base/resource_registry/rest_client.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+        return requests.request(**kwargs)
+
+    def _get_filter_query_params(self, filters: dict):
+        if filters is None:
+            return ""
+
+        query_string = []
+        for k, v in filters.items():
+            query_string.append(f"{k}={v}")
```
Comment (relrod) [inline ansible_base/resource_registry/rest_client.py]: Do we need to urlencode these?

File: test_app/tests/resource_registry/test_resources_api_rest_client.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+from crum import impersonate
+from django.conf import settings
+from django.contrib.auth import get_user_model
+
+from ansible_base.resource_registry.models import Resource, service_id
+from ansible_base.resource_registry.rest_client import ResourceAPIClient
+
+
+@pytest.fixture
+def system_user():
```
Comment (relrod) [inline test_app/tests/resource_registry/test_resources_api_rest_client.py]: Does the one in `conftest.py` not work in this case? If not, can we fix that one and not duplicate?

**Review by relrod — COMMENTED** [review]
Love all those tests!

File: test_app/tests/resource_registry/test_resources_api_rest_client.py (hunk 4 above)
Comment (newswangerd) [reply test_app/tests/resource_registry/test_resources_api_rest_client.py]: oh, i didn't even realize we had one. I think I need the `with impersonate`, but I'll try it.

File: ansible_base/resource_registry/rest_client.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        logger.info(f"Making {method} request to {url}.")
+
+        kwargs = {**self.requests_auth_kwargs, "method": method, "url": url, "verify": self.verify_https}
+
+        if data:
+            kwargs["json"] = data
+
+        return requests.request(**kwargs)
+
+    def _get_filter_query_params(self, filters: dict):
```
Comment (relrod) [inline ansible_base/resource_registry/rest_client.py]: I probably should have thought of this in the first review, but maybe it makes more sense to just pass the filters in as `params` to `requests.request()` and not construct the querystring ourselves:

```python
In [5]: x = requests.request(url='https://example.com', params={'foo': 'bar', 'color': 'green'}, method='get')

In [6]: x.url
Out[6]: 'https://example.com/?foo=bar&color=green'
```

I think this handles encoding for us, too.

File: pyproject.toml [hunk 6]
```diff
@@ -98,7 +98,7 @@ legacy_tox_ini = """
         -r{toxinidir}/requirements/requirements_all.txt
         -r{toxinidir}/requirements/requirements_dev.txt
     allowlist_externals = sh
-    commands = sh -c 'make postgres && pytest -n auto --cov=. --cov-report=xml:coverage.xml --cov-report=html --cov-report=json --cov-branch {env:ANSIBLE_BASE_TEST_DIRS:test_app/tests} {posargs}'
+    commands = sh -c 'make postgres && pytest -n auto --cov=. --cov-report=xml:coverage.xml --cov-report=html --cov-report=json --cov-branch {env:ANSIBLE_BASE_TEST_DIRS:test_app/tests} {posargs}; make stop-postgres'
```
Comment (relrod) [inline pyproject.toml]: This might need to be rebased out, it was removed in #192.

**Review by relrod — APPROVED** [review]
One inline that should probably be the way we do the querystring generation. And needs a rebase to fix the pyproject.toml

Approved pending those changes.

File: ansible_base/resource_registry/rest_client.py (hunk 5 above)
Comment (newswangerd) [reply ansible_base/resource_registry/rest_client.py]: huh. I didn't know you could do that. Thanks!

**Pull request merged**